tuna.go

```diff
@@ -311,6 +314,33 @@ func (c *Common) CreateServerConn(force bool) error {
 
 				metadata := c.GetMetadata()
 
+				if filter != nil && !filter.Empty() {
+					if len(filter.BlackList) > 0 {
+						_, exists := filter.BlackList[metadata.IP]
+						if exists {
+							continue
+						}
+					}
+					if len(filter.WhiteList) > 0 {
+						_, exists := filter.WhiteList[metadata.IP]
+						if !exists {
+							continue
+						}
+					}
+					if !filter.Location.Empty() {
+						loc, err := getLocationFromIP2C(metadata.IP)
+						if err != nil {
+							log.Println(err)
+							continue RandomSubscriber
+						}
+						if loc.Code != filter.Location.Country {
+							log.Printf("node %s from %s is non-compliant", metadata.IP, loc.Country)
```

Comment:
We should remove log on L337 and L340. Otherwise there will be a lot of spam logs...